Added Chroot SimProcedures

Aydalanche opened the pull request.

ltfish commented on angr/procedures/posix/chroot.py:
```diff
@@ -0,0 +1,20 @@
+            return -1
+        
+        self.state.fs.mount('/', angr.SimHostFilesystem(newPath))
+        return 0
```
Please add a blank line at the end of the file.

ltfish commented on tests/test_chroot.py:
```diff
@@ -0,0 +1,24 @@
+        print("Error while trying to open the file")
+
+if __name__ == "__main__":
+    test_chroot()
```
Same here. A blank line at the end of the file.

ltfish commented on tests/test_chroot.py:
```diff
@@ -0,0 +1,24 @@
+import os
+test_location = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries', 'tests')
+
+def test_chroot():
```
Please switch to `unittest.TestCase`-style test cases.

ltfish commented on tests/test_chroot.py:
```diff
@@ -0,0 +1,24 @@
+        stat = simgr.deadended[0].posix.fstat(fd)
+        print(f"File's Stat Size: {stat.st_size}")
+    except:
+        print("Error while trying to open the file")
```
A test case should fail by raising exceptions or assertion failures when success conditions are not met.

Aydalanche commented:
I updated the code to add what you requested and I removed the try and except part of the code from the test file so that it will actually fail if the file or filesystem can't be found.

ltfish commented:
Merged. Thanks!
